src/Util.elm

```diff
@@ -0,0 +1,181 @@
+
+import Http
+import Parser as P exposing ((|.), (|=), Parser, Step)
+
+
+buildErrorMessage : Http.Error -> String
+buildErrorMessage httpError =
+    case httpError of
+        Http.BadUrl message ->
+            message
+
+        Http.Timeout ->
+            "Server is taking too long to respond. Please try again later."
+
+        Http.NetworkError ->
+            "Unable to reach server."
+
+        Http.BadStatus statusCode ->
+            "Request failed with status code: " ++ String.fromInt statusCode
+
+        Http.BadBody message ->
+            message
+
+
+
+{-
+   - Email parser
+   -- valid email should contain valid parts, separated by "@", "."
+   -- ConstructEmail "dooshanstevanovic" "gmail" "com"
+-}
+
+
+type Email
+    = Email String
+
+
+type alias ConstructEmail =
+    { beforeEt : String, afterEt : String, afterDot : String }
+
+
+beforeEtParser : Parser String
+beforeEtParser =
+    P.succeed (\s -> s)
+        |= (P.chompWhile Char.isAlphaNum
+                |> P.getChompedString
+                |> P.andThen
+                    (\beforeEt ->
+                        if String.isEmpty beforeEt || not (isLowerCase beforeEt) then
+                            P.problem "Invalid email"
+
+                        else
+                            P.succeed beforeEt
+                    )
+           )
+
+
+afterEtParser : Parser String
+afterEtParser =
+    P.succeed (\s -> s)
+        |= (P.chompWhile Char.isAlphaNum
+                |> P.getChompedString
+                |> P.andThen
+                    (\afterEt ->
+                        if String.isEmpty afterEt || not (isLowerCase afterEt) then
+                            P.problem "Invalid email"
+
+                        else
+                            P.succeed afterEt
+                    )
+           )
+
+
+afterDotParser : Parser String
+afterDotParser =
+    P.succeed (\s -> s)
+        |= (P.chompWhile Char.isAlphaNum
+                |> P.getChompedString
+                |> P.andThen
+                    (\afterDot ->
+                        if String.isEmpty afterDot || not (isLowerCase afterDot) then
+                            P.problem "Invalid email"
+
+                        else
+                            P.succeed afterDot
+                    )
+           )
+
+
+emailParser : Parser ConstructEmail
+emailParser =
+    P.succeed
+        (\beforeEt afterEt afterDot -> ConstructEmail beforeEt afterEt afterDot)
+        |= beforeEtParser
+        |. P.symbol "@"
+        |= afterEtParser
+        |. P.symbol "."
+        |= afterDotParser
+
+
+parseEmail : String -> Result String Email
+parseEmail email =
+    case P.run emailParser email of
+        Err _ ->
+            Err "Invalid email"
+
+        Ok _ ->
+            Ok (Email email)
+
+
+
+{-
+   - Phone parser
+   -- "+" sign on beggining
+   -- "12345678910" 11 digits string after
```

Comment:
Yeah, true, requirements wasn't too specific, so i saw what could come from Scrive API documentation. Agree, in real world phone numbers comes in many forms and sizes, I guess than it would be up to us to set some rules and boundaries when we write our application. That would be my answer: If we would have hard requirements, let's say: plus sign then country code then phone blah blah -> Would go with parser, if its very rudimental: "+" and any amount of numbers, think regex can work easily.